xline/src/storage/db.rs

```diff
@@ -194,6 +203,10 @@ pub enum WriteOp<'a> {
     PutAppliedIndex(u64),
     /// Put a lease to lease table
     PutLease(PbLease),
+    /// Put a compacted revision into meta table
+    PutCompactRevision(i64),
+    /// Delete a key-value pair from kv table
+    DeleteKeyValue(&'a [u8]),
```

Comment:
`PutKeyValue` use `Revision` as key, `DeleteKeyValue` use `&'a [u8]` as key, it looks weird, can we unify them?

Reply:
I'm afraid not, because `Revision` cannot convert to a bytes reference directly. Let's take a look:
```suggestion
/// Buffered Write Operation
#[derive(Debug, Clone)]
#[non_exhaustive]
pub enum WriteOp<'a> {
    ......
    /// Delete a key-value pair from kv table
    DeleteKeyValue(&'a Revision),
    ......
}

fn flush_ops(&self, ops: Vec<WriteOp>) -> Result<(), ExecuteError> {
      ...
      for op in ops {
          let wop = match op {
              ......
              WriteOp::DeleteKeyValue(rev) => {
                  let key = rev.encode_to_vec();    // `key` does not live long enough
                  WriteOperation::new_delete(KV_TABLE, key.as_ref())
              }
              ... 
      }
      self.engine
          .write_batch(wr_ops, false)
          .map_err(|e| ExecuteError::DbError(format!("Failed to flush ops, error: {e}")))?;
      Ok(())
  }
```

Reply:
Maybe just use `Revision` without reference? `Revision` implements `Copy`

Reply:
Use a buffer like `del_lease_key_buffer` when call `flush_ops`? or use `Vec<u8>` as key when put key value

Reply:
Done